test/test_apply.jl

```diff
@@ -36,19 +36,4 @@ st = coherentstate(FockBasis(1), 1.4)
 
 @test QuantumOpticsBase.apply!(st, 1, sOp).data == (sOp*projector(st)).data
 
-# Test SuperOperator with IncompatibleBases
-b1 = FockBasis(1)
-b2 = FockBasis(2)
-
-k1 = coherentstate(b1, 0.39)
-k2 = coherentstate(b2, 1.4)
-op = projector(k1, k2')
-sOp2 = spre(op)
-
-try
-    QuantumOpticsBase.apply!(st, 1, sOp2)
-catch e
-    @test typeof(e) <: QuantumInterface.IncompatibleBases
-end
```

Comment:
Any reason to remove this test? Should it not simply test for the new error you have?

Reply:
The test wouldn't be for `apply!` anymore since the error is now thrown when calling `spre(op)` which is already tested in `test_superoperators.jl` so I didn't think it made sense have such a test in this file.

Reply:
Let's still keep them -- down the line something might be modified further, and a mistake might be caught only by these tests. Generally, avoid removing tests, even if they are redundant -- at some point they might not have been redundant and we do not want to risk further refactoring in the future making them necessary again.

Reply:
I just reworked this test to now pass but without removing it. I left the previously tested for error commented out in the spirit of avoiding removing tests.